modules/oe_theme_content_event/src/Plugin/ExtraField/Display/DetailsExtraField.php

```diff
@@ -152,41 +142,69 @@ protected function getRenderableDates(ContentEntityInterface $entity): array {
    *   Content entity.
    */
   protected function addRenderableLocation(array &$build, ContentEntityInterface $entity): void {
-    if ($entity->get('oe_event_venue')->isEmpty()) {
-      return;
-    }
-
-    $venue = $entity->get('oe_event_venue')->entity;
-    if (!$venue instanceof VenueInterface) {
-      return;
+    if (!$entity->get('oe_event_venue')->isEmpty() && $entity->get('oe_event_venue')->entity instanceof VenueInterface) {
+      $address = $this->getVenueInlineAddress($entity->get('oe_event_venue')->entity);
+      if (!empty($address)) {
+        // Only display locality and country, inline.
+        $build['#fields']['items'][] = [
+          'icon' => 'location',
+          'text' => $address,
+        ];
+      }
     }
+  }
 
+  /**
+   * Gets rendered address field from Venue entity.
+   *
+   * @param \Drupal\oe_content_entity_venue\Entity\VenueInterface $venue
+   *   Venue entity.
+   *
+   * @return string|null
+   *   Rendered address field.
+   */
+  protected function getVenueInlineAddress(VenueInterface $venue): ?string {
     $venue = $this->entityRepository->getTranslationFromContext($venue);
-    $cacheability = CacheableMetadata::createFromRenderArray($build);
-
     $access = $venue->access('view', NULL, TRUE);
-    $cacheability->addCacheableDependency($access);
-
     if (!$access->isAllowed()) {
-      return;
+      return NULL;
     }
 
-    $cacheability->addCacheableDependency($venue);
-    $cacheability->applyTo($build);
+    // Bubble cacheability metadata of the entity into the current render
+    // context.
+    $build = $this->entityTypeManager->getViewBuilder('oe_venue')->view($venue);
+    $this->renderer->render($build);
 
-    // If address is empty only return cache metadata, so it can bubble up.
     if ($venue->get('oe_address')->isEmpty()) {
-      return;
+      return NULL;
     }
 
-    // Only display locality and country, inline.
+    $location = [];
     $renderable = $this->entityTypeManager->getViewBuilder('oe_venue')->viewField($venue->get('oe_address'));
-    $build['#fields']['items'][] = [
-      'icon' => 'location',
-      'text' => [
-        '#markup' => $renderable[0]['locality']['#value'] . ', ' . $renderable[0]['country']['#value'],
-      ],
-    ];
+    foreach (['locality', 'country'] as $field) {
+      if (!empty($renderable[0][$field]['#value'])) {
+        $location[] = $renderable[0][$field]['#value'];
+      }
+    }
+
+    return implode(', ', $location);
+  }
+
+  /**
+   * Add event online type to event details.
+   *
+   * @param array $build
+   *   Render array.
+   * @param \Drupal\Core\Entity\ContentEntityInterface $entity
+   *   Content entity.
+   */
+  protected function addRenderableOnlineType(array &$build, ContentEntityInterface $entity) {
```

Comment:
just for good practice
```suggestion
  protected function addRenderableOnlineType(array &$build, ContentEntityInterface $entity): void {
```